### DISCUSSION TITLE
Dim inactive windows

### DISCUSSION THREAD

**Discussion by wesbluemarine** [YaLTeR/niri#75] [Ideas]
This could be an alternative to help focusing without ring but never seen any Wayland compositor with a dimming option.

**Comment by YaLTeR** [comment]
By dim, do you mean increase transparency, or desaturate, or darken?

FWIW we were recently trying a few ways of indicating (un-)focused apps in GNOME: https://discourse.gnome.org/t/window-focus-call-for-testing/13277 We found that "toning down" (I believe this was darkening) unfocused windows did not work well UX-wise. Specifically, imagine that you have documentation open in an unfocused window, or a video. Dimming can drastically affect contrast.

**Reply by wesbluemarine** [reply]
Yes, it's not so great for checking unfocused documents.

**Reply by KiaraGrouwstra** [reply]
i'd configured hyprland to dim my inactive windows.
(there was also a toggle to disable transparency if needed.)

it looked pretty rice-y, and i wouldn't mind similar options here.
i imagine this would go well with #113, as after that:
  - rules affecting active/inactive windows might seem a small leap
  - given such rules, transparency settings would become less scary as well

**Reply by KiaraGrouwstra** [reply]
i'd envision generalizing e.g. `(in)active-color` to say:

```kdl
layout {
  active {
    border {
      color 255 200 127 255
    }
  }
  inactive {
    border {
      color 80 80 80 255
    }
  }
}
```
